notion-server/src/api/mod.rs

```diff
@@ -0,0 +1,30 @@
+use crate::ethereum::Handle;
+
+use rocket::{get, routes, State};
+
+use snafu::Snafu;
+
+/// Shorthand for result types returned from the API server.
+pub type Result<V, E = Error> = std::result::Result<V, E>;
+
+/// Errors arising from the API server.
+#[derive(Debug, Snafu)]
+pub enum Error {}
+
+pub fn run(handle: Handle) -> Result<()> {
+    // TODO: Get some configuration options from command line.
+
+    rocket::ignite()
+        .mount("/", routes![hello])
+        .manage(handle)
```

Comment:
Rocket has a concept of state that it can provide to request handlers. `manage` is how you add that state. Here's [the relevant section of the guide](https://rocket.rs/v0.4/guide/state/) if you'd like some further reading.